src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,46 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+
+public class IntakeSubsystem extends SubsystemBase {
+
+    private final TalonFX intakeMotor;
+
+    private final DigitalInput frontBeamBreak;
+    private final DigitalInput backBeamBreak;
+
+    TalonFXConfiguration INTAKE_MOTOR_CONFIG = new TalonFXConfiguration();
+
+    public IntakeSubsystem() {
+        intakeMotor = new TalonFX(0); 
+        frontBeamBreak = new DigitalInput(0);
+        backBeamBreak = new DigitalInput(0);
+        applyConfigs();
+    }
+
+    private void applyConfigs() {
+        INTAKE_MOTOR_CONFIG.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+        INTAKE_MOTOR_CONFIG.MotorOutput.Inverted = InvertedValue.Clockwise_Positive; //TODO: Change if needed
+        intakeMotor.getConfigurator().apply(INTAKE_MOTOR_CONFIG);
+    }
+
+    public void setIntakeSpeed(double speed) {
+        intakeMotor.set(speed);
+    }
+
+    public boolean isRingAtFrontOfIntake() {
+        return frontBeamBreak.get();
+    }
```

Comment:
i would think about what we want this to be named. The way it currently works, at least how I would think about it is, "if this function returns true, that means the ring is at the front of the intake". However, if this function returns true, it means that the beambreak can see the other side, meaning nothing is obstructing it. When something is in the beambreak, it will return false. Meaning, this method will actually return true when nothing is in the intake, and false when something is. I would negate the .get() method call, so the method returns true when something is obstructing the beam break.

Reply:
You want to `return !frontBeamBreak.get();`

What Oliver was pointing out is that the way a beam break works is it's shooting light from an emitter to a sensor. When the sensor sees light, the `DigitialInput` returns `true`. But, of course, you want to return `true` when there is something in the way of the light.